rm-datasource/src/main/java/io/seata/rm/datasource/sql/druid/postgresql/PostgresqlInsertRecognizer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.rm.datasource.sql.druid.postgresql;
+
+import com.alibaba.druid.sql.ast.SQLExpr;
+import com.alibaba.druid.sql.ast.SQLStatement;
+import com.alibaba.druid.sql.ast.expr.SQLIdentifierExpr;
+import com.alibaba.druid.sql.ast.expr.SQLValuableExpr;
+import com.alibaba.druid.sql.ast.expr.SQLVariantRefExpr;
+import com.alibaba.druid.sql.ast.statement.SQLExprTableSource;
+import com.alibaba.druid.sql.ast.statement.SQLInsertStatement;
+import com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGInsertStatement;
+import com.alibaba.druid.sql.dialect.postgresql.visitor.PGOutputVisitor;
+import io.seata.rm.datasource.sql.SQLInsertRecognizer;
+import io.seata.rm.datasource.sql.SQLParsingException;
+import io.seata.rm.datasource.sql.SQLType;
+import io.seata.rm.datasource.sql.druid.BaseRecognizer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author japsercloud
+ */
+public class PostgresqlInsertRecognizer extends BaseRecognizer implements SQLInsertRecognizer {
+
+    private final PGInsertStatement ast;
+
+    /**
+     * Instantiates a new My sql insert recognizer.
+     *
+     * @param originalSQL the original sql
+     * @param ast the ast
+     */
+    public PostgresqlInsertRecognizer(String originalSQL, SQLStatement ast) {
+        super(originalSQL);
+        this.ast = (PGInsertStatement) ast;
+    }
+
+    @Override
+    public SQLType getSQLType() {
+        return SQLType.INSERT;
+    }
+
+    @Override
+    public String getTableAlias() {
+        return ast.getTableSource().getAlias();
+    }
+
+    @Override
+    public String getTableName() {
+        StringBuffer sb = new StringBuffer();
```

Comment:
please use StringBuilder instead of StringBuffer, by the way, check the order Recognizer

Reply:
fixed